### PR TITLE
rocksdb 7.3.1

### PR DESCRIPTION
**Pull request by alexreg**
This builds and installs the pkgconfig file (`rocksdb.pc`). There is no rule for it in the Makefile generated by CMake, but there *is* in the original Makefile, and since the `gen-pc` rule is not coupled with any of the rest of the file, it makes sense to use that.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
I think it also makes sense to report this upstream so they can fix the inconsistency. And to use pkg-config in the test then.

**Comment by alexreg** [comment]
@SMillerDev It already got reported years ago, and I reminded them recently. But they're very slow to act.

**Comment by alexreg** [comment]
Is there any reason this hasn't been merged yet?

**Comment by alexreg** [comment]
We could wait until next release of rocksdb now that a Cmake way of generating pkgconfig files has been added. Shall we?

**Comment by SMillerDev** [comment]
Sounds good

**Comment by alexreg** [comment]
So, the PR for this just got merged into master. I say we include a patch for now, with a TODO note, and remove it when the next version lands (probably 7.3.0). Just pushed an update to this effect.

File: Formula/rocksdb.rb
Comment (SMillerDev) [inline Formula/rocksdb.rb:115]: That looks like it's a bug since that should already be the case.

Comment (alexreg) [reply Formula/rocksdb.rb:115]: Oh okay, I'll remove that. Should we remove `"-L#{lib}"` too?

Comment (alexreg) [reply Formula/rocksdb.rb]: Seems to build just fine without `"-L#{lib}"`, so I'm guessing it's included by default in `ENV.cxx`.

Comment (alexreg) [reply Formula/rocksdb.rb]: Have a look at my latest push, providing it passes.

**Comment by alexreg** [comment]
Still waiting for RocksDB 7.3.0 release...

**Comment by alexreg** [comment]
@SMillerDev v7.3.1 is out now, and I've updated the PR, so this should be ready to merge.

**Comment by SMillerDev** [comment]
The tests are failing, that will need to be fixed first

**Comment by alexreg** [comment]
> The tests are failing, that will need to be fixed first

Ah, I removed a compiler option I thought was redundant. Should be fine now.

**Comment by iMichka** [comment]
I adjusted the commit message. Should be good to go once the build is done.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**
